packages/flutter/lib/src/services/text_editing.dart

```diff
@@ -57,12 +57,25 @@ class TextRange {
   /// The text before this range.
   String textBefore(String text) {
     assert(isNormalized);
+    if(text.isNotEmpty && start > 0 && start < text.length ){
```

Comment:
nit: spacing should be `if (text.isNotEmpty && start > 0 && start < text.length) {`